ui/pages/swaps/build-quote/build-quote.js

```diff
@@ -392,6 +410,51 @@ export default function BuildQuote({
     fromTokenAddress,
     selectedToToken.address,
   );
+  const isReviewSwapButtonDisabled =
+    tokenFromError ||
+    !isFeatureFlagLoaded ||
+    !Number(inputValue) ||
+    !selectedToToken?.address ||
+    Number(maxSlippage) < 0 ||
+    Number(maxSlippage) > MAX_ALLOWED_SLIPPAGE ||
+    (toTokenIsNotDefault && occurrences < 2 && !verificationClicked);
+
+  // It's triggered every time there is a change in form values (token from, token to, amount and slippage).
+  useEffect(() => {
+    dispatch(clearSwapsQuotes());
+    dispatch(stopPollingForQuotes());
+    const prefetchQuotesWithoutRedirecting = async () => {
+      const pageRedirectionDisabled = true;
+      await dispatch(
+        fetchQuotesAndSetQuoteState(
+          history,
+          inputValue,
+          maxSlippage,
+          metaMetricsEvent,
+          pageRedirectionDisabled,
+        ),
+      );
+    };
+    // Delay fetching quotes until a user is done typing an input value. If they type a new char in less than a second,
+    // we will cancel previous setTimeout call and start running a new one.
+    timeoutIdForQuotesPrefetching = setTimeout(() => {
+      timeoutIdForQuotesPrefetching = null;
```

Comment:
Should we clear the previous timeout here, in addition to setting the variable to null?

Reply:
When this `useEffect` call for checking form changes is triggered again, it's automatically cleared by:
```
return () => clearTimeout(timeoutIdForQuotesPrefetching);
```